### ISSUE TITLE
OPTIONS requests on presigned URL's wrongly check signature, causing failure for client-side requests

### ISSUE THREAD

**Issue by mmezei** [gaul/s3proxy#411]
When a browser based application uses a pre-signed URL (typically generated by its API), the browser issues an OPTIONS request first to check CORS. OPTIONS requests apparently do not include the content-md5 and x-amz-acl headers that are required for successful signature validation. This causes the OPTIONS request to fail with a SignatureDoesNotMatch error, resulting in the browser fail the PUT. Amazon does not check the signature on the OPTIONS request. I was able to see this by sending a bogus signature to Amazon and similarly Digital Ocean.

**Comment by rajivml** [comment]
@mmezei were you able to get pass this issue? am running into a similar issue

**Comment by mmezei** [comment]
Yes. Please see my fork... Let me know if you can't locate it.

**Comment by gaul** [comment]
Sorry I don't know much about CORS.  @snoe925 is this something you could look at?

@mmezei Any reason you are maintaining a fork instead of merging fixes upstream?  https://github.com/Switch-TV/s3proxy/commit/9786ab0bcfd82d6895c7e620cdae8cb1fba2ef29  seems like a plausible fix.

**Comment by mmezei** [comment]
Yes. Tried to contact upstream regarding the issues at the time and could
not make contact for a pull request as author did not seem active at that
time.

**Comment by rajivml** [comment]
thanks @mmezei , this works

**Comment by daveanderson** [comment]
Is this https://github.com/Switch-TV/s3proxy/commit/9786ab0bcfd82d6895c7e620cdae8cb1fba2ef29 fix something that can be applied and included in a release? (I'd rather not run have to run a fork)

**Comment by Wamy-Dev** [comment]
this is still an issue for me. Any possibility of a fix @gaul ?

**Comment by gaul** [comment]
I cherry-picked this but please just submit pull requests in the future.
